spot-ml/src/main/scala/org/apache/spot/utilities/transformation/ProbabilityConverter.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spot.utilities.transformation
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions._
+
+/**
+  * ProbabilityConverter will transform document probability from Double to Float if scaling option is set to 32 bit,
+  * if default or 64 bit is selected, it will just return the same Double probability.
+  *
+  * This abstract class permits the execution of a single path during the entire analysis. Instead of checking what
+  * type should the data be converted, each implementation will know what type should be converted into and converted
+  * back.
+  *
+  */
+
+abstract class ProbabilityConverter extends Serializable {
```

Comment:
Several superficial comments:

- no need for the word probability in the definition of this file or its classes... nothing currently in the file is particular to probabilities and could be applied to any column containing a Seq[Double] or a Seq[Float]

- since we are not actually "scaling" anything, "ScalingType" should be something like "TargetType"